### PR TITLE
split kvmtest-t0 job into two parts for Azure.sonic-mgmt pipeline

### PR DESCRIPTION
**Pull request by ZhaohuiS**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911

#### What is the motivation for this PR?
Currently, Azure.sonic-mgmt pipeline is running for more than 4 hours, especially for kvmtest-t0 job. Split kvmtest-t0 job into two separated jobs, each job runs part of test cases for t0. Reduce the waiting time of PR merge.

#### How did you do it?
Split kvmtest-to job into kvmtest-t0-part1 and kvmtest-t0-part2, split t0 test cases into two jobs.

#### How did you verify/test it?
Submit PR will trigger the pipeline and check the whole running time of pipeline.

### CODE REVIEW

**Comment by ZhaohuiS** [comment]
Two paralleled sub kvmtest t0 jobs both run for about 2h20m, the original kvmtest t0 job usually ran for more than 4h30m.
image

File: .azure-pipelines/run-test-template.yml [hunk 1]
```diff
@@ -83,17 +86,17 @@ steps:
   condition: succeededOrFailed()
 
 - publish: $(Build.ArtifactStagingDirectory)/kvmdump
-  artifact: sonic-buildimage.kvmtest.${{ parameters.tbtype}}.memdump@$(System.JobAttempt)
+  artifact: sonic-buildimage.kvmtest.${{ parameters.tbtype}}${{ parameters.section}}.memdump@$(System.JobAttempt)
```
Comment (wangxin) [inline .azure-pipelines/run-test-template.yml:89]: Need a whitespace before `}}`

File: .azure-pipelines/run-test-template.yml [hunk 2]
```diff
@@ -83,17 +86,17 @@ steps:
...
   displayName: "Archive sonic kvm memdump"
   condition: failed()
 
 - publish: $(Build.ArtifactStagingDirectory)/logs
-  artifact: sonic-buildimage.kvmtest.${{ parameters.tbtype }}.log@$(System.JobAttempt)
+  artifact: sonic-buildimage.kvmtest.${{ parameters.tbtype }}${{ parameters.section}}.log@$(System.JobAttempt)
```
Comment (wangxin) [inline .azure-pipelines/run-test-template.yml:94]: Need a whitespace before }}

File: .azure-pipelines/run-test-template.yml [hunk 3]
```diff
@@ -83,17 +86,17 @@ steps:
...
   displayName: "Archive sonic kvm logs"
   condition: succeededOrFailed()
 
 - task: PublishTestResults@2
   inputs:
     testResultsFiles: '$(Build.ArtifactStagingDirectory)/logs/**/*.xml'
-    testRunTitle: kvmtest.${{ parameters.tbtype }}
+    testRunTitle: kvmtest.${{ parameters.tbtype }}${{ parameters.section}}
```
Comment (wangxin) [inline .azure-pipelines/run-test-template.yml:101]: Need a whitespace before }}

File: azure-pipelines.yml [hunk 4]
```diff
@@ -31,6 +31,22 @@ stages:
         ptf_name: ptf_vms6-1
         tbtype: t0
         vmtype: ceos
+        section: part-1
+
+  - job:
+    pool: sonictest
+    displayName: "kvmtest-t0-part2"
+    timeoutInMinutes: 400
```
Comment (wangxin) [inline azure-pipelines.yml:39]: Probably we can consider reducing this timeout value now.

File: azure-pipelines.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ stages:
   jobs:
   - job:
     pool: sonictest
-    displayName: "kvmtest-t0"
+    displayName: "kvmtest-t0-part1"
     timeoutInMinutes: 400
```
Comment (wangxin) [inline azure-pipelines.yml:24]: Probably we can consider reducing this timeout value now.

Comment (ZhaohuiS) [reply azure-pipelines.yml]: Good point, update it to 200.

File: azure-pipelines.yml (hunk 4 above)
Comment (ZhaohuiS) [reply azure-pipelines.yml]: Good point, update it to 200.

File: .azure-pipelines/run-test-template.yml (hunk 3 above)
Comment (ZhaohuiS) [reply .azure-pipelines/run-test-template.yml]: Done

File: .azure-pipelines/run-test-template.yml (hunk 2 above)
Comment (ZhaohuiS) [reply .azure-pipelines/run-test-template.yml]: Done

File: .azure-pipelines/run-test-template.yml (hunk 1 above)
Comment (ZhaohuiS) [reply .azure-pipelines/run-test-template.yml]: Done

**Review by wangxin — APPROVED** [review]

File: azure-pipelines.yml [hunk 6]
```diff
@@ -31,6 +46,20 @@ stages:
...
         vmtype: ceos
+        section: part-2
+
+  - job:
+    displayName: "kvmtest-t0"
+    timeoutInMinutes: 400
+    dependsOn:
+      - t0-part1
+      - t0-part2
+    condition: |
```
Comment (xumia) [inline azure-pipelines.yml:57]: I guess the condition is always().
You can define a variable to indicate the job is successful.
```
variables:
  DEP_RESULT: ${{ convertToJson(dependencies.*.result) }}
steps:
  script: |
    ### exist successful or failed based on $(DEP_RESULT)
```

Comment (ZhaohuiS) [reply azure-pipelines.yml]: Thanks for the reminder. always is a good way, we can also add continueOnError: true in two sub jobs to let job kvmtest-t0 run every time even sub job fails.
Conditionally run a step such as the following code doesn't work: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#conditional-insertion
```
     steps:
     - ${{ if and(eq(variables.resultOfPart1, 'Succeeded'), eq(variables.resultOfPart2, 'Succeeded')) }}:
       - script: 'echo "Run kvmtest-t0 successfully!"'
     - ${{ else }}:
	- script: 'echo "Run kvmtest-t0 failed!"; exit 1'
```

So change to use script:
```
    steps:
    - script: |
        if [ $(resultOfPart1) == "Succeeded" ] && [ $(resultOfPart2) == "Succeeded" ]; then
          echo "Both job kvmtest-t0-part1 and kvmtest-t0-part2 are passed."
          exit 0
        else
          echo "Either job kvmtest-t0-part1 or job kvmtest-t0-part2 failed! Please check the detailed information."
          exit 1
        fi
```

**Review by wangxin — APPROVED** [review]

**Pull request merged**